### PR TITLE
Integration of arctic embeddings with pyserini

### PR DESCRIPTION
**Pull request by UShivani3**
Adds pre-built indexes for Snowflake's arctic models

### CODE REVIEW

**Review by lintool — CHANGES_REQUESTED** [review]
`add_md5.py` is empty... why?
You need another readme file for embed-m?

File: pyserini/encode/_base.py [hunk 1]
```diff
@@ -76,6 +76,7 @@ def load_encoded_queries(cls, encoded_query_name: str):
     @staticmethod
     def _load_embeddings(encoded_query_dir):
         df = pd.read_pickle(os.path.join(encoded_query_dir, 'embedding.pkl'))
+        df['text'] = df['text'].apply(lambda x: x.strip())
```
Comment (lintool) [inline pyserini/encode/_base.py:79]: maybe let's not change this... because it might break something else?

File: docs/experiments-arctic.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# Pyserini: Reproducing Arctic Results
```
Comment (lintool) [inline docs/experiments-arctic.md:1]: Let's rename this file `experiments-msmarco-v2.1-arctic.md`

File: pyserini/encode/_base.py (hunk 1 above)
Comment (lintool) [reply pyserini/encode/_base.py:79]: Yes, please revert this change.

File: pyserini/encoded_query_info.py [hunk 3]
```diff
@@ -1034,5 +1034,15 @@
         "size (bytes)": 135674,
         "total_queries": 49,
         "downloaded": False
+    },
```
Comment (lintool) [inline pyserini/encoded_query_info.py:1037]: Let's remove this for now.

File: pyserini/prebuilt_index_info.py [hunk 4]
```diff
@@ -5181,7 +5230,20 @@ def add_lucene_index_info(enum, info, name=None, readme=None):
         "documents": 5183,
         "downloaded": False,
         "texts": "beir-v1.0.0-scifact.flat"
-    }
+    },
+    "beir-v1.0.0-nq.arctic-embed-m-v1.5": {
```
Comment (lintool) [inline pyserini/prebuilt_index_info.py:5234]: Let's remove this for now.

File: pyserini/resources/index-metadata/faiss-flat.msmarco-v2.1-doc.arctic-embed-l.20240824.README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
+# msmarco-v2.1-arctic-embed-l
```
Comment (lintool) [inline pyserini/resources/index-metadata/faiss-flat.msmarco-v2.1-doc.arctic-embed-l.20240824.README.md:1]: We're missing a README for arctic-m?

File: pyserini/util.py [hunk 6]
```diff
@@ -69,6 +69,7 @@ def download_url(url, save_dir, local_filename=None, md5=None, force=False, verb
     if not local_filename:
         filename = url.split('/')[-1]
         filename = re.sub('\\?dl=1$', '', filename)  # Remove the Dropbox 'force download' parameter
+        filename = re.sub(r'\?.*$', '', filename)
```
[the remark this answers is not in the document]
Comment (lintool) [reply pyserini/util.py:72]: Please revert.

File: scripts/arctic/temp.sh [hunk 7]
```diff
@@ -0,0 +1,22 @@
+mv ArguAna faiss-flat.beir-v1.0.0-arguana.arctic-embed-m-v1.5.20241031
```
Comment (lintool) [inline scripts/arctic/temp.sh:1]: I don't think you mean to check this in?

**Review by lintool — CHANGES_REQUESTED** [review]
Also, remove `scripts/arctic/add_md5.py`

**Comment by UShivani3** [comment]
> `add_md5.py` is empty... why? You need another readme file for embed-m?

removed

**Review by lintool — APPROVED** [review]

**Pull request merged**